lib/Mount/FolderStorageManager.php

```diff
@@ -99,24 +134,38 @@ public function getBaseStorageForFolderSeparateStorageLocal(
 		if ($init) {
 			$storage->getScanner()->scan('');
 		}
+		return $storage;
+	}
 
-		if ($folder && $folder->acl && $user) {
-			$aclManager = $this->aclManagerFactory->getACLManager($user);
-			return new ACLStorageWrapper([
-				'storage' => $storage,
-				'acl_manager' => $aclManager,
-				'in_share' => $inShare,
-				'storage_id' => $storage->getCache()->getNumericStorageId(),
-			]);
-		} else {
-			return $storage;
+	private function getBaseStorageForFolderSeparateStorageObject(
+		int $folderId,
+	): IStorage {
+		$objectStoreConfig = $this->primaryObjectStoreConfig->getObjectStoreConfiguration($this->getObjectStorageKey($folderId));
+
+		if ($objectStoreConfig['arguments']['multibucket']) {
+			$objectStoreConfig['arguments']['bucket'] = $this->getObjectStorageBucket($folderId, $objectStoreConfig);
+		}
+
+		$objectStore = $this->primaryObjectStoreConfig->buildObjectStore($objectStoreConfig);
+		$arguments = array_merge($objectStoreConfig['arguments'], [
+			'objectstore' => $objectStore,
+		]);
+		$arguments['storageid'] = 'object::groupfolder:' . $folderId . '.' . $objectStore->getStorageId();
+
+		$storage = new ObjectStoreStorage($arguments);
+
+		if (!$storage->file_exists('files')) {
+			$storage->mkdir('files');
+			$storage->mkdir('trash');
+			$storage->mkdir('versions');
 		}
```

Comment:
Each of the folders should be checked for existence. If creating `files` works bu the other two don't, then it will run into problems later as the directories are not recreated.